scripts/ci/Dockerfile

```diff
@@ -55,16 +62,16 @@ ENV NODE_PATH "$NVM_DIR/versions/node/$NODE_VERSION/lib/node_modules"
 ENV PATH "$NVM_DIR/versions/node/$NODE_VERSION/bin:/bin:$HOME/.local/bin:/root/.local/bin:$PATH"
 RUN mkdir -p $NVM_DIR
 
-# Install Node Version Manager, Node Package Manager, and the "frontend" packages
+# Install Node Version Manager, NodeJS and the packages for CI (package-lock.json) 
 RUN curl -o- $NVM_INSTALLER | bash
 RUN /bin/bash -c "source $NVM_DIR/nvm.sh && \
     nvm install $NODE_VERSION && \
     nvm use --delete-prefix $NODE_VERSION && \
     npm config set prefix '$NPM_CACHE' && \
     echo 'export PATH=$PATH' >> '$USER_HOME/.profile' && \
     source $USER_HOME/.profile && \
-    npm install -g npm@latest && \
-    npm cache -g clean --force"
+    npm cache -g clean --force && \
+    npm ci --cache=$NPM_CACHE"
```

Comment:
From what I read, it seems that install will still fetch from the internet and fallback to cache if not available. There were talks about using `--prefer-offline` to reverse this lookup. Can you also give this a try and provide some comparison times between with and without this PR.

Reply:
@christianvogt Are you sure? I just took a look at [npm cache](https://docs.npmjs.com/cli/v10/commands/npm-cache) and how the [cache flag works](https://docs.npmjs.com/cli/v10/using-npm/config#cache) and I haven't been able to find any reference about that. I only see that they do a internet check first to ensure cache integrity (which i think is needed) so I would assume this is the right ting to do.
You can even do that setting the cache with `npm set cache ...`, but I think this is more straightforward.
But I don't get the part of cleaning the global cache though, if we are creating the container and we only use a custom cache folder I would assumen the global folder will be empty right?
I've built this image and it seems to be working just fine but it will be interesting checking the running difference as @christianvogt is mentioning.

Reply:
I didn't get a chance yet to test this as I was just doing some googling on the topic. So take what I've said as with a grain of salt :)

After reading your response and looking a bit more I think this is correct.

At the end of the day I don't expect the npm install to be a big contributor to the overall time.

Reply:
On a side note. When we get to actually running e2e tests, we don't need to install all the packages of the frontend; rather only those needed to run cypress. We may look at creating a sub package strictly for cypress.